src/pages/login/index.js

```diff
@@ -0,0 +1,60 @@
+const login = function() {
+  const $root = document.querySelector("#root");
+
+  const $loginButton = flatButton.render("Log in", true, "login");
+  const $signinButton = flatButton.render("Sign up", false, "signup");
+  const $logoCollabcode = logoCollabcode.render();
+  const $titleCollabcode = titleCollabcode.render("Hello!");
+  const $logoWrapper = logoWrapper.render($logoCollabcode, $titleCollabcode);
+  const $formLogin = formLogin.render();
+
+  $root.insertAdjacentHTML("beforeend", $loginButton);
+  $root.insertAdjacentHTML("beforeend", $signinButton);
+  $root.insertAdjacentHTML("beforeend", $logoWrapper);
+  $root.insertAdjacentHTML("beforeend", $formLogin);
+
+  //Adicionando as validações da página
+  const $form = document.querySelector("#form");
+
+  const $email = document.querySelector("#email");
+  const $emailError = document.querySelector("span[for='email']");
+
+  const $password = document.querySelector("#password");
+  $password.setAttribute("minlength", "8");
+  $password.setAttribute("maxlength", "8");
+  const $passwordError = document.querySelector("span[for='password']");
+
+  $email.addEventListener("input", () => {
+    if ($email.validity.valid) {
+      $emailError.innerHTML = "";
+      $emailError.classList.remove("-active");
+    }
+  });
+
+  $password.addEventListener("input", () => {
+    if ($password.validity.valid) {
+      $passwordError.innerHTML = "";
+      $passwordError.classList.remove("-active");
+    }
+  });
+
+  $form.addEventListener("submit", event => {
+    event.preventDefault();
+
+    if ($email.validity.valid && $password.validity.valid) {
+      location.hash = `#/game`;
+      location.reload(true);
+      return true;
+    }
+
+    if (!$email.validity.valid) {
+      $emailError.innerHTML = "E-mail inválido!";
+      $emailError.classList.add("-active");
+    }
+
+    if (!$password.validity.valid) {
+      $passwordError.innerHTML = "Sua senha deve conter 8 caracteres!";
+      $passwordError.classList.add("-active");
+    }
```

Comment:
Aqui é mais uma comentário de backend do que de front em si. A galera recomenda colocar geralmente uma mensagem mais genérica "E-mail e/ou senha inválidos" ao invés de uma específica para não dar dicas para um possível invasor de que ele "acertou" o e-mail por exemplo. É lógico que se sua aplicação for muito bem testada e tiver uma segurança muito boa, não tem problema nenhum mostrar as informações deste jeito. É só algo pra você ter em mente.

Reply:
De fato a mensagem deveria ser mais genérica, mas acabei colocando um span para validação de cada campo. O ideal era ter um span para exibir todas as inconsistências e no caso dessa tela, apenas uma mensagem genérica.